datafusion/physical-plan/src/repartition/mod.rs

```diff
@@ -1414,4 +1398,23 @@ mod tests {
 
         Ok(())
     }
+
+    /// Create vector batches
+    fn create_vec_batches(schema: &Schema, n: usize) -> Vec<RecordBatch> {
+        let batch = create_batch(schema);
+        let mut vec = Vec::with_capacity(n);
+        for _ in 0..n {
+            vec.push(batch.clone());
+        }
+        vec
+    }
+
+    /// Create batch
+    fn create_batch(schema: &Schema) -> RecordBatch {
```

Comment:
This is a rather strange function, it takes a schema, but then proceeds to assume the schema...